mllib/src/main/scala/org/apache/spark/ml/tree/treeParams.scala

```diff
@@ -360,7 +362,9 @@ private[ml] trait RandomForestParams extends TreeEnsembleParams {
     "The number of features to consider for splits at each tree node." +
       s" Supported options: ${RandomForestParams.supportedFeatureSubsetStrategies.mkString(", ")}",
     (value: String) =>
-      RandomForestParams.supportedFeatureSubsetStrategies.contains(value.toLowerCase))
+      RandomForestParams.supportedFeatureSubsetStrategies.contains(value.toLowerCase)
+      || (try { value.toInt > 0 } catch { case _ : Throwable => false })
```

Comment:
This could be simplified using a regex. I think the following pattern should work: `"0?\\.\\d+|1.0+|\\d+"`